atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -86,16 +86,18 @@ void WebFrame::SetSpellCheckProvider(mate::Arguments* args,
   web_frame_->view()->setSpellCheckClient(spell_check_client_.get());
 }
 
-void WebFrame::RegisterURLSchemeAsSecure(const std::string& scheme) {
-  // Register scheme to secure list (https, wss, data).
-  blink::WebSecurityPolicy::registerURLSchemeAsSecure(
-      blink::WebString::fromUTF8(scheme));
-}
+void WebFrame::RegisterURLSchemeAsPrivileged(const std::string& scheme) {
+  // Register scheme to privileged list (https, wss, data, chrome-extension)
+  using RegisterFunction = void (*)(const blink::WebString&);
+  RegisterFunction register_functions[] = {
+    blink::WebSecurityPolicy::registerURLSchemeAsSecure,
+    blink::WebSecurityPolicy::registerURLSchemeAsBypassingContentSecurityPolicy,
+    blink::WebSecurityPolicy::registerURLSchemeAsAllowingServiceWorkers
+  };
 
-void WebFrame::RegisterURLSchemeAsBypassingCsp(const std::string& scheme) {
-  // Register scheme to bypass pages's Content Security Policy.
-  blink::WebSecurityPolicy::registerURLSchemeAsBypassingContentSecurityPolicy(
-      blink::WebString::fromUTF8(scheme));
+  blink::WebString privileged_scheme(blink::WebString::fromUTF8(scheme));
+  for (RegisterFunction func : register_functions)
```

Comment:
I don't see a reason to call the APIs like this, we still need to enumerate every function and the code doesn't look cleaner.